src/main/kotlin/uk/gov/dluhc/printapi/mapper/StatusMapper.kt

```diff
@@ -0,0 +1,32 @@
+package uk.gov.dluhc.printapi.mapper
+
+import uk.gov.dluhc.printapi.database.entity.Status.DISPATCHED
+import uk.gov.dluhc.printapi.database.entity.Status.IN_PRODUCTION
+import uk.gov.dluhc.printapi.database.entity.Status.NOT_DELIVERED
+import uk.gov.dluhc.printapi.database.entity.Status.PRINT_PROVIDER_DISPATCH_FAILED
+import uk.gov.dluhc.printapi.database.entity.Status.PRINT_PROVIDER_PRODUCTION_FAILED
+import uk.gov.dluhc.printapi.database.entity.Status.PRINT_PROVIDER_VALIDATION_FAILED
+import uk.gov.dluhc.printapi.database.entity.Status.VALIDATED_BY_PRINT_PROVIDER
+import uk.gov.dluhc.printapi.messaging.models.ProcessPrintResponseMessage.StatusStep
+import uk.gov.dluhc.printapi.database.entity.Status as StatusEntityEnum
+import uk.gov.dluhc.printapi.messaging.models.ProcessPrintResponseMessage.Status as StatusModelEnum
+
+class StatusMapper {
```

Comment:
I expected using `MapStruct` interface?

Reply:
From what @abdullah-sanver-valtech mentioned in his reply on the original comment on this, I think he tried using MapStruct and couldnt get it to work (or something?), so resorted to writing it like this
Personally I prefer this written like this vs MapStruct "magic" - I can read this code really easily and grok what its doing. MapStruct is just voodoo and wizardry to me! 🤣

Reply:
```
@Mapper
abstract class StatusMapper {

    @ValueMapping(target = "VALIDATED_BY_PRINT_PROVIDER", source = "PROCESSED")
    @ValueMapping(target = "IN_PRODUCTION", source = "IN_MINUS_PRODUCTION")
    @ValueMapping(target = "DISPATCHED", source = "DISPATCHED")
    @ValueMapping(source = MappingConstants.ANY_UNMAPPED, target = MappingConstants.THROW_EXCEPTION)
    abstract fun toStatusEntityEnumOnSuccessStatus(statusStep: StatusStep): StatusEntityEnum

    @ValueMapping(target = "NOT_DELIVERED", source = "NOT_MINUS_DELIVERED")
    @ValueMapping(target = "PRINT_PROVIDER_VALIDATION_FAILED", source = "PROCESSED")
    @ValueMapping(target = "PRINT_PROVIDER_PRODUCTION_FAILED", source = "IN_MINUS_PRODUCTION")
    @ValueMapping(target = "PRINT_PROVIDER_DISPATCH_FAILED", source = "DISPATCHED")
    abstract fun toStatusEntityEnumOnFailedStatus(statusStep: StatusStep): StatusEntityEnum
```

Reply:
@NathanRussellValtech @abdullah-sanver-valtech 

```
@Mapper
interface StatusMapper {

    @ValueMapping(target = "VALIDATED_BY_PRINT_PROVIDER", source = "PROCESSED")
    @ValueMapping(target = "IN_PRODUCTION", source = "IN_MINUS_PRODUCTION")
    @ValueMapping(target = "DISPATCHED", source = "DISPATCHED")
    @ValueMapping(source = MappingConstants.ANY_UNMAPPED, target = MappingConstants.THROW_EXCEPTION)
    fun toStatusEntityEnumOnSuccessStatus(statusStep: StatusStep): StatusEntityEnum

    @ValueMapping(target = "NOT_DELIVERED", source = "NOT_MINUS_DELIVERED")
    @ValueMapping(target = "PRINT_PROVIDER_VALIDATION_FAILED", source = "PROCESSED")
    @ValueMapping(target = "PRINT_PROVIDER_PRODUCTION_FAILED", source = "IN_MINUS_PRODUCTION")
    @ValueMapping(target = "PRINT_PROVIDER_DISPATCH_FAILED", source = "DISPATCHED")
    fun toStatusEntityEnumOnFailedStatus(statusStep: StatusStep): StatusEntityEnum
```

Reply:
I dont think that does the same thing does it? We want a single mapper method that takes the `StatusStep` and the `Status`
What are the target enums eg: `PRINT_PROVIDER_VALIDATION_FAILED` - they don't exist and are not part of our design in this respect 🤔 

I think the version Abdullah has is easier to read by far 👍 💯

Reply:
> What are the target enums eg: `PRINT_PROVIDER_VALIDATION_FAILED` - they don't exist and are not part of our design in this respect 🤔 

Actually they are, ignore me 😁 

I still think What Abdullah is proposing is better though 👍